pkg/controller/node/ipam/cloud_cidr_allocator.go

```diff
@@ -105,6 +106,12 @@ func NewCloudCIDRAllocator(client clientset.Interface, cloud cloudprovider.Inter
 			if newNode.Spec.PodCIDR == "" {
 				return ca.AllocateOrOccupyCIDR(newNode)
 			}
+			// Even if PodCIDR is assigned, but NetworkUnavailable condition is
+			// set to true, we need to process the node to set the condition.
+			_, cond := v1node.GetNodeCondition(&newNode.Status, v1.NodeNetworkUnavailable)
+			if cond != nil && cond.Status != v1.ConditionTrue {
```

Comment:
This should be ConditionFalse right?

Reply:
Actually `cond == nil || cond.Status == v1.ConditionFalse` (unless I'm understanding it wrong)?

Reply:
Yes - good catch.

Reply:
Fixed